app/qml/form/editors/MMFormPhotoEditor.qml

```diff
@@ -110,26 +112,17 @@ MMFormPhotoViewer {
     }
   }
 
-  MessageDialog { // TODO: We might want to redesign this dialogue
+  MMRemovePhotoDialog {
     id: imageDeleteDialog
 
     property string imagePath
 
-    title: qsTr( "Remove photo reference" )
-    text: qsTr( "Also permanently delete photo from device?" )
-
-    buttons: MessageDialog.Yes | MessageDialog.No | MessageDialog.Cancel
-
-    onButtonClicked: function( clickedButton ) {
-      if ( clickedButton === MessageDialog.Yes ) {
-        internal.imageSourceToDelete = imageDeleteDialog.imagePath
-        root.editorValueChanged( "", false ) // Shouldn't this be true?
-      }
-      else if ( clickedButton === MessageDialog.No ) {
-        root.editorValueChanged( "", false ) // Shouldn't this be true?
-      }
-
-      close()
+    onDeleteImage: {
+      internal.imageSourceToDelete = imageDeleteDialog.imagePath
+      root.editorValueChanged( "", true )
```

Comment:
I would keep `root.editorValueChanged( "", false ) // Shouldn't this be true?` or are you sure you know it should be true?